manager/src/main/java/org/openremote/manager/map/MapService.java

```diff
@@ -78,6 +87,31 @@ public class MapService implements ContainerService {
     protected Map<String, ObjectNode> mapSettings = new HashMap<>();
     protected Map<String, ObjectNode> mapSettingsJs = new HashMap<>();
 
+    public ObjectNode saveMapConfig(Object mapConfiguration) throws Exception {
+        LOG.log(Level.INFO, "Saving mapsettings.json");
+        this.mapConfig.putNull("options");
+        this.mapSettings.clear();
+        ObjectNode mapSettings = loadMapSettingsJson(mapSettingsPath);
+        try {
+            OutputStream out = new FileOutputStream(new File(mapSettingsPath.toUri()));
+            ObjectMapper mapper = new ObjectMapper();
+            mapSettings.putPOJO("options", mapConfiguration);
+            mapper
+                    .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY)
+                    .setSerializationInclusion(JsonInclude.Include.NON_NULL)
+                    .enable(SerializationFeature.INDENT_OUTPUT);
+
+            out.write(mapper.writeValueAsString(mapSettings).getBytes());
+            out.close();
+            this.setData();
+        } catch (IOException exception) {
```

Comment:
Best practice is to use try-resource with anything that implements `Closeable`; see:

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html